linera-execution/solidity/linera.sol

```diff
@@ -3,13 +3,53 @@
 pragma solidity ^0.8.0;
 
 // Precompile keys:
-// 0: try_call_application
-// 1: try_query_application
-// 2: send_message
-// 3: message_id
-// 4: message_is_bouncing
-
+// (0,0): chain_id
+// (0,1): application_creator_chain_id
+// (0,2): chain_ownership
+// (0,3): read data blob
+// (0,4): assert data blob exists
+// (1,0): try_call_application
+// (1,1): validation round
+// (1,2): send_message
+// (1,3): message_id
+// (1,4): message_is_bouncing
+// (2,0): try_query_application
 library Linera {
+    function bcs_deserialize_offset_bytes32(uint256 pos, bytes memory src) internal pure returns (uint256, bytes32) {
+        bytes32 dest;
+        assembly {
+            dest := mload(add(add(src, 0x20), pos))
+        }
+        uint256 new_pos = pos + 32;
```

Comment:
Both `0x20` and `32` are magic numbers. Please add at least comments explaining what they are for.

Reply:
In that case 0x20 = 32. But yes, I can document that.